client/asset/btc/btc.go

```diff
@@ -460,12 +480,21 @@ func BTCCloneWallet(cfg *BTCCloneCFG) (*ExchangeWallet, error) {
 // newWallet creates the ExchangeWallet and starts the block monitor.
 func newWallet(cfg *BTCCloneCFG, btcCfg *dexbtc.Config, node rpcClient) *ExchangeWallet {
 	// If set in the user config, the fallback fee will be in conventional units
-	// per kB, e.g. BTC/kB. Translate that to sats/B.
+	// per kB, e.g. BTC/kB. Translate that to sats/byte.
 	fallbackFeesPerByte := toSatoshi(btcCfg.FallbackFeeRate / 1000)
 	if fallbackFeesPerByte == 0 {
 		fallbackFeesPerByte = cfg.DefaultFallbackFee
 	}
-	cfg.Logger.Tracef("Fallback fees set at %d %s/vbyte", fallbackFeesPerByte, cfg.WalletInfo.Units)
+	cfg.Logger.Tracef("Fallback fees set at %d %s/vbyte",
+		fallbackFeesPerByte, cfg.WalletInfo.Units)
+
+	// If set in the user config, the fee rate limit will be in units of BTC/KB.
+	// Convert to sats/byte.
+	feesLimitPerByte := toSatoshi(btcCfg.FeeRateLimit / 1000)
+	if feesLimitPerByte == 0 {
+		feesLimitPerByte = cfg.DefaultFeeRateLimit
+	}
```

Comment:
You might consider rejecting fee rates `1e-8 <= r < 1e-5 `, since the user is clearly trying to set something, and is probably trying to set a low value, but these will truncate to zero, resulting in the default value being used.

Reply:
Good point about rounding to zero.  @amassarwi You'll have to rebase to resolve some conflicts before merge.

Reply:
Done

Reply:
Sorry to unresolve this since the PR was so close to done, but I think it just got a little hairier than necessary with the resolution to this that was pushed.

There are two things that I'm not content with now: (1) The new `dcrCfg.FeeRateLimit != 0 && dcrCfg.FeeRateLimit < 1/(smallestUnit/1000)` check could instead be _inferred_ after computing `feesLimitPerByte`, and (2) the error return seems avoidable with the reasonable resolution of rounding up to 1 sat/byte for fee rates <1e-5 DCR/kB.

For dcr, consider:

```go
	feesLimitPerByte := uint64(defaultFeeRateLimit)
	if dcrCfg.FeeRateLimit > 0 {
		feesLimitPerByte = toAtoms(dcrCfg.FeeRateLimit / 1000)
		if feesLimitPerByte == 0 {
			feesLimitPerByte = 1 // round up very low specified rates
		}
	}
```

IMO, we just want to avoid using the default when the specified rate was truncated/rounded to zero when converting units.  I don't have a concern about rounding up (to 1 atom/byte) as a resolution when the specified rate is below 1e-5 DCR/kB.  But I would like to avoid duplicate math with the `smallestUnit` const and the added error return.

Personally, I don't think any user will have a problem with rounding up to the nearest atom/byte, but we're bloating lots of code to add this error return, and adding some error prone and redundant computations.

Reply:
@buck54321 round up to 1 {atom,sat}/byte instead of lots more error return handling sit ok with you?

Reply:
OK, we just spoke offline and decided that actually it's safest to meet `if feesLimitPerByte == 0 {` with an error instead of rounding up to 1.  But let's clean up the math and the variable name as suggested in the other comment.

Reply:
So we did chat a little on matrix. The question on the table is two-fold. 1. Should values lower than `1e-5` be rounded up to `1e-5`, and 2) should it be an error for values < `1e-8`?

I think the safest bet is to assume nothing and error for invalid values < `1e-5`, which are arguably invalid. On the other hand, we could error only for values < `1e-8`, and round values `1e-8 <= r < 1e-5` up to the minimum valid value, `1e-5`, but we are making an assumption about the user's intention in that case, so my preference is to simply error for values `v != 0 && v < 1e-5`.

Reply:
We agree about the handling to error for `v != 0 && v < 1e-5`, although it's best to keep the check bulletproof by simply checking as in my suggested code, but instead of defaulting to 1, return an error.  Small final change.

Reply:
👍 

will do